src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result: GroupsMap<T>, item: T) => {
+    const groupKey = String(item[key]);
+
+    if (!result[groupKey]) {
+      result[groupKey] = [];
+    }
+    result[groupKey].push(item);
+
+    return result;
+  }, {});
```

Comment:
Consider explicitly defining the type for the initial value of the reduce function. Right now, it's inferred as `{}` which is not very descriptive. You can use `as GroupsMap<T>` to cast the initial value to the correct type, ensuring the object structure is consistent with the `GroupsMap<T>` type.